hs_core/views/__init__.py

```diff
@@ -192,12 +199,24 @@ def update_key_value_metadata(request, shortkey, *args, **kwargs):
     if resource_mode is not None:
         request.session['resource-mode'] = 'edit'
 
-    if is_update_success:
-        messages.success(request, "Metadata update successful")
+    if isinstance(request, HttpRequest):
+        if is_update_success:
+            messages.success(request, "Metadata update successful")
+        else:
+            messages.error(request, err_message)
+
+        return HttpResponseRedirect(request.META['HTTP_REFERER'])
+
     else:
-        messages.error(request, err_message)
+        if is_update_success:
+            return HttpResponse(status=200)
+        else:
+            return HttpResponse(status=400)
 
-    return HttpResponseRedirect(request.META['HTTP_REFERER'])
+
+@api_view(['POST'])
+def update_key_value_metadata_public(request, pk):
+    return update_key_value_metadata(request, pk)
```

Comment:
Since this function is now handling rest call and UI call without much common code, would it be better to handle rest call in its own function and the leave the original function unchanged?